proposals/NNNN-high-level-interface-foundation.md

```diff
@@ -0,0 +1,421 @@
+
+@property (nonatomic, copy) NSArray<SDLTTSChunk *> *initialPrompt;
+@property (nonatomic, copy) SDLSamplingRate samplingRate;
+@property (nonatomic, copy) SDLBitsPerSample bitsPerSample;
+@property (nonatomic, copy) SDLAudioType audioType;
+@property (nonatomic) BOOL muteMediaSource;
+
+@property (nonatomic) SDLOnAudioDataBlock onAudioDataBlock;
+
+- (void)dismiss; // sends an EndAudioPassThru in order to stop audio input
+
+@end
+```
+
+#### SDLScrollableMessageController
+
+Similar to `Alert` this view controller takes care about the RPC `ScrollableMessage`.
+
+```objc
+@interface SDLScrollableMessageController : SDLModalViewController
+
+@property (nonatomic, copy) NSArray<SDLSoftButton *> *buttons;
+
+@end
+```
+
+#### SDLSliderController
+
+This controller matches very closely to the view [`UISlider`](https://developer.apple.com/documentation/uikit/uislider). For consitency it makes more sense to treat it as a modal controller. In order to work with the RPC `Slider` it is important to provide the selected value back to the caller. As of UIKit this is not done by using a custom completion handler but simply provide the result in a property of the modal view controller. As per mobile API the property `Slider.sliderFooter` is an array used for two purposes. Either it's a footer (single item) or a list of names representing slider values (number of items must match `.numTicks`). The mobile API allows `numTicks` to be between 2 and 26. With the high level abstraction the modal controller can manage much more value ranges than 1-26. Examples are ranges with negative (or more values than available) which are mapped (and scaled) internally to the available range. Future versions of the mobile API could increase the range which would be managed by the slider controller.
+
+```objc
+@interface SDLSliderController : SDLModalViewController
+
+@property (nonatomic, copy) NSString *title; // mapped with Slider.sliderHeader
+@property (nonatomic, copy) NSArray<NSString *> *valueLables; // overrides control of SDLModalViewController.message 
+
+// minimumValue and maximumValue map to .numTicks
+@property (nonatomic) NSInteger minimumValue;
+@property (nonatomic) NSInteger maximumValue;
+
+// maps to `Slider.position`
+@property (nonatomic) NSInteger value;
+
+@end
+```
+
+#### SDLInteractionController
+
+Together with the `SDLChoiceSetManager` the interaction controller will take care about most of the painful work to deal with choice sets and interactions.
+
+In order to support the app developer a class called `SDLChoiceSet` is used which comes with a choice set manager. The choice set management will be proposed separately as it should be contained in the management layer.
+
+- If `.manualInteraction` is set the controller performs with interaction mode MANUAL_ONLY.
+- If `.voiceInteraction` is set the controller  performs with interaction mode VOICE_ONLY.
+- If both parameters are seth the controller perform with interaciton mode BOTH.
```

Comment:
Please update to "If both parameters are set the controller performs with interaction mode BOTH."